Automattic/woocommerce-payments
# Pass blocks data to woopay

Fixes #5482 

#### Changes proposed in this Pull Request
Changes introduced here extract Blocks settings data fields from Automatewoo, Mailchimp and Mailpoet and then pass them to WooPay when initiating the checkout. For Automatewoo and Mailchmp, the data fields are extracted by registering the blocks on a standalone [Integration Registry](https://github.com/woocommerce/woocommerce-blocks/blob/dea2050a69d0faaa45cf113ec40ad0b39c95613f/docs/third-party-developers/extensibility/checkout-block/integration-interface.md#L22). I didn't find a good way to follow the same approach for Mailpoet, so its data are extracted directly from its Settings instance without registering the plugin.

This data extraction mechanism is a stopgap until we have a more permanent API based approach to fetch these data for WooPay.








#### Testing instructions




Regression tests
* Make sure Automatewoo, Mailchimp and Mailpoet are *not* installed in the merchant.
* Add some products to the cart and go through WooPay checkout.
* Make sure checkout is functional and no errors show up in WC logs

Test passed data fields
* Install WooCommerce Blocks
* Install and activate Automatewoo
* Go to blocks checkout page
* Open devtools and note the fields in `wcSettings.automatewoo_data`
* Add a breakpoint [here](https://github.com/Automattic/woocommerce-payments/blob/483a1da05d09f35a44c07d2a8f1839872f8c3fe2/includes/class-wc-payments.php#L1140)
* Add a product to the cart and go to the checkout page.
* Use an already WooPay registered email to initiate the WooPay OTP prompt
* Go through OTP prompt, and note the value of [blocks_data](https://github.com/Automattic/woocommerce-payments/blob/483a1da05d09f35a44c07d2a8f1839872f8c3fe2/includes/class-wc-payments.php#L1138) at the breakpoint
* Make sure its values are same as the ones noted in `wcSettings.automatewoo_data`
* Follow the above steps again for Mailchimp and Mailpoet and match them against the values in `wcSettings.mailchimp-newsletter_data` and `wcSettings.mailpoet_data`

-------------------


**Post merge**
